File: tests/cases/enforce/factory/lua/inputs/src/orders/checkout.lua
local Receipt = require("orders.receipt")

local CheckoutService = {}
CheckoutService.__index = CheckoutService

-- Delegates construction. Silent.
function CheckoutService:checkout(total)
  return self.receipts:for_total(total)
end

-- The violation: `T.new(...)` outside the designated factory. A bare
-- `Receipt(...)` would NOT count - in Lua that is `__call`, not a constructor.
function CheckoutService:quick_receipt(total)
  return Receipt.new(total)
end

return CheckoutService

File: tests/cases/enforce/factory/python/inputs/src/orders/checkout.py
from orders.factory import ReceiptFactory
from orders.receipt import Receipt


class CheckoutService:
    def __init__(self, receipts: ReceiptFactory) -> None:
        self.receipts: ReceiptFactory = receipts

    def checkout(self, total: float) -> Receipt:
        """Delegates construction. Silent."""
        return self.receipts.for_total(total)

    def quick_receipt(self, total: float) -> Receipt:
        """The violation: a shortcut around the designated factory."""
        return Receipt(total)
